tests/php/test-class-wc-connect-api-client.php

```diff
@@ -1,37 +1,58 @@
 <?php
 
+/**
+ * Unit test for WC_Connect_API_Client_Live
+ */
 class WP_Test_WC_Connect_API_Client extends WC_Unit_Test_Case {
 
+	/** @var WC_Connect_API_Client_Live $api_client */
 	protected $api_client;
+	/** @var WC_Product $product */
 	protected $product;
 
+	/**
+	 * Undocumented function
+	 */
 	public static function setupBeforeClass() {
-		require_once( dirname( __FILE__ ) . '/../../classes/class-wc-connect-api-client.php' );
-		require_once( dirname( __FILE__ ) . '/../../classes/class-wc-connect-api-client-live.php' );
+		require_once dirname( __FILE__ ) . '/../../classes/class-wc-connect-api-client.php';
+		require_once dirname( __FILE__ ) . '/../../classes/class-wc-connect-api-client-live.php';
 
```

Comment:
PHPCS outputs on line 19:

**Blank line found at end of control structure (WordPress.WhiteSpace.ControlStructureSpacing.BlankLineBeforeEnd)**

Reply:
@tommyshellberg Interesting, thanks for pointing this out. Mine doesn't have this error. My `phpcs -i` is `The installed coding standards are PEAR, Zend, PSR2, MySource, Squiz, PSR1, PSR12, PHPCompatibility, PHPCompatibilityParagonieRandomCompat, PHPCompatibilityParagonieSodiumCompat, PHPCompatibilityWP, WooCommerce-Core, WordPress, WordPress-Extra, WordPress-Docs, WordPress-Core and Extendables` 

I do see `WordPress` in there. One thing came to mind is I haven't updated my rulesets for a year. Maybe mine is outdated.

In the `phpcs.xml.dist`, the `ruleset` is simply `<ruleset name="WordPress Coding Standards">`. I think maybe it is using whatever our system has? In that case, I should add a dev dependency in the repo so we all use the latest one. 

I am going to take a look at this.

Reply:
@tommyshellberg I ran `./vendor/bin/phpcs -vvv tests/php/test-class-wc-connect-api-client.php > csoutput` to dump the output into the `csoutput` file. I wanted to see what rules it is using. 

It looks like it does use the composer package rules; woo core and [wpcs's 2.3.0](https://github.com/woocommerce/woocommerce-sniffs/blob/b72b7dd2e70aa6aed16f80cdae5b1e6cce2e4c79/composer.json#L20).

```
Processing ruleset woocommerce-services/phpcs.xml.dist
	=> set config value minimum_supported_wp_version: 4.7
	=> set config value testVersion: 5.6-
	Processing rule "WooCommerce-Core"
		=> woocommerce-services/vendor/woocommerce/woocommerce-sniffs/src/WooCommerce-Core/ruleset.xml
		* rule is referencing a standard using ruleset path; processing *
		Processing ruleset woocommerce-services/vendor/woocommerce/woocommerce-sniffs/src/WooCommerce-Core/ruleset.xml
			Processing rule "WooCommerce"
...
Processing rule "WordPress"
	=> woocommerce-services/vendor/wp-coding-standards/wpcs/WordPress/ruleset.xml
...
```

In the log, I looked for `WordPress.WhiteSpace.ControlStructureSpacing` and I saw this:
```
Processing rule "Squiz.Strings.ConcatenationSpacing"
	=> woocommerce-services/vendor/squizlabs/php_codesniffer/src/Standards/Squiz/Sniffs/Strings/ConcatenationSpacingSniff.php
	=> property "spacing" set to "1"
	=> property "ignoreNewlines" set to "true"
Processing rule "WordPress.WhiteSpace.ControlStructureSpacing"
	=> woocommerce-services/vendor/wp-coding-standards/wpcs/WordPress/Sniffs/WhiteSpace/ControlStructureSpacingSniff.php
```

It looks like the WP 2.3.0 rule has Squiz spacing sniff [set to ignore new lines](https://github.com/WordPress/WordPress-Coding-Standards/blob/7da1894633f168fe244afc6de00d141f27517b62/WordPress-Core/ruleset.xml#L213). The next sniff runs the WordPress `ControlStructureSpacingSniff` and effectively ignoring that new line check.

I am curious what yours look like? Is your `phpcs` using the `vendor/` rules or is yours using the global rules? Can you try running `composer run phpcs tests/php/test-class-wc-connect-api-client.php` or `./vendor/bin/phpcs tests/php/test-class-wc-connect-api-client.php` and see if you pass the tests please?

Reply:
I get the same message for both `composer run phpcs tests/php/test-class-wc-connect-api-client.php` and `./vendor/bin/phpcs tests/php/test-class-wc-connect-api-client.php`.:

![Screen Shot 2020-10-28 at 12 05 14 PM](https://user-images.githubusercontent.com/6723003/97428137-d7c5bc80-1915-11eb-880e-6af7a19ca394.png)

Reply:
Mm that's strange. It also flagged the `Squiz.Commenting.FileComment.Missing` error but we ignored that in the `phpcs.xml.dist` `<rule ref="Squiz.Commenting.FileComment.Missing">`. We can do `./vendor/bin/phpcs -vvv tests/php/test-class-wc-connect-api-client.php` to see what rules it's using. 

> I don't think we should worry too much about it right now, if we want to pursue it further it could be during a cooldown period.

Sounds good. Merging.